bentoml/_internal/bento/bento.py

```diff
@@ -88,23 +88,45 @@ def _Bento_set_fs(bento: "Bento", _: t.Any, new_fs: "FS") -> "FS":
 @attr.define(repr=False, auto_attribs=False)
 class Bento(StoreItem):
     _tag: Tag = attr.field()
-    _fs: "FS" = attr.field(on_setattr=_Bento_set_fs)
+    __fs: "FS" = attr.field(on_setattr=_Bento_set_fs)
 
-    info: "BentoInfo"
+    _info: "BentoInfo"
 
     _model_store: ModelStore
     _doc: t.Optional[str] = None
 
+    _flushed: bool = False
+
+    _export_ext: str = "bento"
+
     def __init__(self, tag: Tag, bento_fs: "FS", info: "BentoInfo"):
         self._tag = tag
-        self._fs = _Bento_set_fs(self, None, bento_fs)
+        self.__fs = _Bento_set_fs(self, None, bento_fs)
         self.info = info
         self.validate()
 
     @property
     def tag(self) -> Tag:
         return self._tag
 
+    @property
+    def _fs(self) -> "FS":
+        return self.__fs
+
+    @property
+    def info(self) -> "BentoInfo":
+        self._flushed = False
```

Comment:
Why accessing `.info` will mark self dirty? Incase we use `bento.info['options'] = xxx`?

Reply:
Should we allow user to change the `info` at all? or is it only used internally?   It seems the only way user can access to a Bento instance is from `get`, `list` and `build`, where we probably don't want user to change the `info`

Reply:
I think freezing it is probably correct. Should we do the same for models?

Reply:
Yes I also think we should be freezing it here

Reply:
About models, it depends on the model creation Api we expose to users.

Currently we have:
```
model = Model.create()
# dump model files to model.path
# assign metadata to model.metadata
model.save()
```
In this case, it must be mutable so user can set metadata, options, and custom_objects.

I think one of the idea we explored a bit earlier was to provide a `with Model.create() as model:` block, in that case, we can freeze everything after the `with` block.

Reply:
Will do this in a separate PR.

Reply:
sounds good!